include/exadg/solvers_and_preconditioners/utilities/petsc_operation.h

```diff
@@ -96,9 +90,35 @@ apply_petsc_operation(VectorType &                                           dst
       dst.local_element(i) = ptr[i];
     }
 
-    ierr = VecRestoreArray(vector_dst, &ptr);
+    ierr = VecRestoreArray(petsc_vector_dst, &ptr);
     AssertThrow(ierr == 0, ExcPETScError(ierr));
   }
+}
+
+/*
+ *  This function wraps the copy of a PETSc object (sparse matrix,
+ *  preconditioner) with a dealii::LinearAlgebra::distributed::Vector,
+ *  allocating a PETSc vectors and then calling the other function
+ */
+template<typename VectorType>
+void
+apply_petsc_operation(VectorType &                                           dst,
+                      VectorType const &                                     src,
+                      MPI_Comm const &                                       petsc_mpi_communicator,
+                      std::function<void(PETScWrappers::VectorBase &,
+                                         PETScWrappers::VectorBase const &)> petsc_operation)
+{
+  Vec vector_dst, vector_src;
```

Comment:
here I would write `petsc_vector_dst` for consistency with the other `apply_petsc_operation()` function.

Reply:
Done in 0f0e2b69.